### PR TITLE
Add c8 for coverage and GH action

### PR DESCRIPTION
**Pull request by MathieuLamiot**
# Description

Add code coverage report to Codacy with c8 and a Github action

### User documentation

NA

### Technical documentation

NA

## Type of change
*Delete options that are not relevant.*

- [x] New feature (non-breaking change which adds functionality).
- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] Enhancement (non-breaking change which improves an existing functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as before).

## New dependencies
c8
Codacy

## Risks
NA

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .idea
 node_modules
 /dist
-package-lock.json
\ No newline at end of file
+package-lock.json
+coverage
```
Comment (jeawhanlee) [inline .gitignore:5]: Why was this added here?

Comment (MathieuLamiot) [reply .gitignore:5]: package-lock.json: I think it's better not to store it in GitHub, just like composer.lock. So that, when running npm install on a fresh checkout, dependencies from package.json are checked instead of relying on the intermediary lock file.
Since I added a dependency in my commit, I saw it was part of the changes, which made me realize it probably shouldn't.

coverage is added because this folder is generated when running test coverage. This is not data we want to store on GitHub.

**Review by jeawhanlee — APPROVED** [review]
